presto-native-execution/README.md

```diff
@@ -96,17 +85,90 @@ checks.  When we need one it can be added.
+  * Open any h or cpp file in the editor and select `Enable ClangFormat` by clicking `4 spaces` rectangle in the status bar (bottom right) which is next to `UTF-8` bar.
+
+### Using local Presto environment on MacOs (with optional IntelliJ and CLion)
+Running Presto Coordinator + Worker on mac
+* Note that everything below can be done w/o using IDEs by running command line commands (not in this readme).
+* Run 'HiveQueryRunnerExternal' from IntelliJ and wait until it started (`======== SERVER STARTED ========` in the log output).
+* Scroll up the log output and find `Discovery URL http://127.0.0.1:50555`. The port is 'random' with every start.
+* Copy that port (or the whole URL) to the `discovery.uri` field in `presto/presto-native-execution/etc/config.properties` for the worker to Discover the Coordinator.
+* In CLion run "presto_server" module. Connection success will be indicated by `Announcement succeeded: 202` line in the log output.
+* In command line from presto root folder run the presto client:
+  * `java -jar presto-cli/target/presto-cli-*-executable.jar --catalog hive --schema tpch`
+* You can start from `show tables;` and `describe table;` queries and execute more queries you want.
+
+### Running Integration (End to End or E2E) Tests on MacOs (with optional IntelliJ and CLion)
+* Note that everything below can be done w/o using IDEs by running command line commands (not in this readme).
+* Open a test file which has the test(s) you want to run in IntelliJ from `presto/presto-native-execution/src/test/java/com/facebook/presto/nativeworker` path.
+* Click the green arrow to the left of the test class line of code and chose if you want to Run or Debug. This will run all tests in this class. 
+* Alternatively click the green arrow to the left of the test class' test method line of code and chose if you want tor Run or Debug. This will run all tests only in this class's member.
+* The framework will launch single Coordinator and four native workers to test-run the queries.
+* Similarly, the unit tests of Velox and presto_cpp can be run from CLion.
+
+### Advancing Velox's version
+* For presto_cpp to use a newer Velox's version from the presto repository root:
+  * `git -C presto-native-execution/velox checkout main`
+  * `git -C presto-native-execution/velox pull`
+  * `git add presto-native-execution/velox`
+  * Build and run tests (including E2E) to ensure everything works.
+  * Submit a PR, get it approved and merged.
+
+### Creating PRs for presto/presto-native-execution/
+* Submit PRs as usual following Presto repository guidelines (https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines).
+* Ot top of it please add `[native]` prefix in the title for PRs modifying anything in `presto-native-execution`. 
+* PRs that only change files in `presto-native-execution` should be approved by a Code Owner (`team-velox` https://github.com/orgs/prestodb/teams/team-velox) to have merging enabled.  
```

Comment:
I think proper syntax for links is `[team-velox](https://github.com/orgs/prestodb/teams/team-velox)`

Reply:
Got that. :)
I actually wasn't going to make it a proper hyperlink.